### ISSUE TITLE
Encapsulate peer dialing, connections into a `tower::Discover` instance.

### ISSUE THREAD

**Issue by hdevalence** [ZcashFoundation/zebra#47]
Tower provides a generic service discovery mechanism which is composable with other Tower functionality (e.g., automatically wrapping services with load information).  We should encapsulate peer creation (both outgoing and incoming) in a type that implements the `Discover` trait.

**Comment by hdevalence** [comment]
This should use #45 internally when spinning up new `Service`s.

**Comment by hdevalence** [comment]
Per slack discussion:  The current `connect` stub code builds a `Discover` instance using a `ServiceStream`; because channels are streams, the Discover implementation could just be a ServiceStream wrapping a `mpsc` channel.  One sender could be the `PeerListener` #46, while another could be a task with #45 and a `PeerConnector`, which the `PeerSet` can signal to.

**Comment by hdevalence** [comment]
Per https://github.com/ZcashFoundation/zebra2-maybe/issues/45#issuecomment-542336084, the dependency graph looks like:
```
.
└── 47
    ├── 45
    │   └── 49
    │       └── 65
    └── 46
```

**Comment by hdevalence** [comment]
Closed by #78
